### PR TITLE
CI: Build fuzzers with shared cargo target dir

### PR DESCRIPTION
**Pull request by Mrmaxmeier**
This might speed up CI a bit, but requires >15 GB in disk space so might also just fail.

### CODE REVIEW

**Comment by VTCAKAVSMoACE** [comment]
Possible alternative: sccache drastically speeds up the builds. It seems that having a common target dir isn't great since they're always overwriting each other.

**Comment by domenukk** [comment]
Noone got sccache to work properly for libafl

**Comment by VTCAKAVSMoACE** [comment]
image

It's working just fine for me locally. What was the issue? There's a lot of things that get cache missed, sure, but it's still a significant speed up.

**Comment by domenukk** [comment]
Think disk size in CI (maybe?) Don't remember

**Comment by domenukk** [comment]
See the discussion in #579

**Comment by domenukk** [comment]
Cc @vanhauser-thc

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -95,10 +95,10 @@ jobs:
     - name: Check each feature
       # Skipping `python` as it has to be built with the `maturin` tool
       # `agpl`, `nautilus` require nightly
-      # `sancov_pcguard_edges` are tested seperately
+      # `sancov_pcguard_edges` is tested seperately
       run: cargo hack check --each-feature --exclude-features=agpl,nautilus,python,sancov_pcguard_edges,arm,aarch64,i386,be
     - name: Check nightly features
-      run: cargo +nightly hack check --each-feature --include-features=agpl,nautilus
```
Comment (domenukk) [inline .github/workflows/build_and_test.yml]: Hack actually checks different permutations of features, which would be nice to keep (?)

**Comment by vanhauser-thc** [comment]
trying to us on on here: https://github.blog/2022-09-01-github-actions-introducing-the-new-larger-github-hosted-runners-beta/

**Comment by domenukk** [comment]
Speed seems to have doubled, accepting for now (athough I might eventually bring `hack --depth=2` back)

**Comment by domenukk** [comment]
Feel free to open a new PR for more changes if you feel like it, thx! :)

**Pull request merged**